src/main/java/subway/stations/SubwayStations.java

```diff
@@ -30,6 +30,73 @@ public int getObjectId() {
         }
     }
 
+    public List<Station> dijkstrasAlgorithm(Station source, Station destination){
+
+        Set<Station> visitedStations = new HashSet<>();
+        Set<Station> unvisitedStations = new HashSet<>();
+
+        Station currentSource = source;
+        int shortestDistance = Integer.MAX_VALUE;
+
+        while(!(currentSource.connections.contains(destination))){
+            visitedStations.add(currentSource);
+            unvisitedStations.addAll(currentSource.connections);
+
+            for (Station station : currentSource.connections) {
+                station.previous = currentSource;
```

Comment:
You shouldn't be setting these values each time.

Reply:
How else should I keep track of the parent for tracing?

Reply:
Sorry.
What I meant was that you should only set `previous` **IF** you are replacing `distanceFromSource`. and you shouldn't be doing that unless the new distanceFromSource is less than the current value.